server/api/routes/subscriptions.js

```diff
@@ -48,4 +49,26 @@ router.delete('/:subscriptionUuid', async (req, res) => {
   }
 });
 
+router.get('/update', async (req, res) => {
+  const { user_id } = req.session.userInfo;
+  const lateDueDates = [];
+  try {
+    const allSubscriptions = await getSubscriptionsByUserId(user_id);
+    for (let i = 0; i < allSubscriptions.length; i += 1) {
+      const subscription = allSubscriptions[i];
+      const updatedSubscription = await updateNextDueDate(subscription.dueDate, subscription.subscriptionUuid);
+      const { dueDate, name } = updatedSubscription;
+      if (updatedSubscription) {
+        if (dueDate.lateDueDate) {
+          lateDueDates.push({ name, date: dueDate.lateDueDate });
+        }
+      }
+    }
```

Comment:
this may be worth doing in parallel since you're just updating each subscription individually and doesn't rely on other subscriptions finishing their update first. You can use `await Promise.all()` and then create an array of all of the updates then filter out those that don't have a lateDueDate.

another option I explored is the use `await Promise.all()` and then a `reduce()` in order to just pass back the updated data:

```javascript
  const { user_id } = req.session.userInfo;
  try {
    const allSubscriptions = await getSubscriptionsByUserId(user_id);
    const updatedSubscriptions = await Promise.all(allSubscriptions.map(({dueDate, subscriptionUuid}) =>
      updateNextDueDate(dueDate, subscriptionUuid)));

    const lateDueDates = updatedSubscriptions.reduce((lateSubs, subscription) => {
      if (subscription?.dueDate?.lateDueDate) {
        lateSubs.push({ name: subscription.name, date: subscription.dueDate.lateDueDate});
      }
      return lateSubs;
    }, []);

    res.status(200).json({ lateDueDates });
  } catch(error) {
    console.log('Error: ', error);
  }
```